devito/symbolics.py

```diff
@@ -83,90 +90,190 @@ def dse_indexify(expr):
     return expr.xreplace(replacements)
 
 
-def dse_cse(expr):
+def dse_rewrite(expr, mode='basic'):
     """
-    Perform common subexpression elimination on sympy expressions.
-
-    :param expr: sympy equation or list of equations on which CSE is performed.
-
-    :return: A list of the resulting equations after performing CSE
+    Transform expressions to create time-invariant computation.
     """
-    expr = expr if isinstance(expr, list) else [expr]
 
-    temps, stencils = cse(expr, numbered_symbols("temp"))
+    if mode == True:
+        return Rewriter(expr).run(mode='advanced')
+    elif mode in ['basic', 'advanced']:
```

Comment:
True. I'll add a couple of lines. Basically:
basic -> CSE
advanced -> CSE + Factorization

We default to ``advanced`` for obvious reasons we want to have both
Will fix this !